Sources/ConnectionPoolModule/PoolStateMachine.swift

```diff
@@ -673,9 +677,20 @@ struct PoolStateMachine<
         let requests = self.requestQueue.pop(max: availableContext.info.availableStreams)
         if !requests.isEmpty {
```

Comment:
what happens, if we have min connections = 5, but are coming out of circuit breaker and don't have anything in the queue?